dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/security/SessionController.java

```diff
@@ -0,0 +1,119 @@
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.webapi.controller.security;
+
+import static org.hisp.dhis.security.Authorities.ALL;
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.hisp.dhis.common.DhisApiVersion;
+import org.hisp.dhis.common.HashUtils;
+import org.hisp.dhis.common.OpenApi;
+import org.hisp.dhis.feedback.NotFoundException;
+import org.hisp.dhis.security.RequiresAuthority;
+import org.hisp.dhis.user.User;
+import org.hisp.dhis.user.UserDetails;
+import org.hisp.dhis.user.UserService;
+import org.hisp.dhis.webapi.mvc.annotation.ApiVersion;
+import org.springframework.security.core.session.SessionInformation;
+import org.springframework.security.core.session.SessionRegistry;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * @author Morten Svanæs <msvanaes@dhis2.org>
+ */
+@OpenApi.Document(
+    entity = User.class,
+    classifiers = {"team:platform", "purpose:support"})
+@RestController
+@RequestMapping("/api/sessions")
+@RequiredArgsConstructor
+@ApiVersion({DhisApiVersion.DEFAULT, DhisApiVersion.ALL})
+public class SessionController {
+
+  private final SessionRegistry sessionRegistry;
+  private final UserService userService;
+
+  @GetMapping(produces = APPLICATION_JSON_VALUE)
+  @RequiresAuthority(anyOf = ALL)
+  public Map<String, String> listAllSessions() throws NotFoundException {
+    return listAllUserSessions().stream()
+        .collect(
+            Collectors.toMap(
+                s -> HashUtils.hashSHA1(s.getSessionId().getBytes()),
+                s ->
+                    ((UserDetails) s.getPrincipal()).getUsername()
+                        + (s.isExpired() ? ", (inactive)" : ", (active)")));
+  }
+
+  private List<SessionInformation> listAllUserSessions() throws NotFoundException {
+    List<SessionInformation> allSessions = new ArrayList<>();
+    List<User> allUsers = userService.getAllUsers();
+    for (User user : allUsers) {
+      allSessions.addAll(
+          sessionRegistry.getAllSessions(userService.createUserDetails(user.getUid()), true));
+    }
+    return allSessions;
+  }
+
+  @PostMapping(value = "/{username}/invalidate", produces = APPLICATION_JSON_VALUE)
```

Comment:
Same as below, isn't using `@DeleteMapping("/{username}")` more fitting?
Also it does not `produces = APPLICATION_JSON_VALUE`. Like a typical delete it doesn't return anything, which should use `@ResponseStatus(HttpStatus.NO_CONTENT)